src/rest/Rest.tsx

```diff
@@ -228,6 +228,40 @@ export const Rest = {
     };
     return postFetcher(URLs.EXECUTION_HISTORY_MODULES, data);
   },
+
+  getDebugExecutions: function () {
+    return getFetcher(URLs.DEBUG_EXECUTIONS);
+  },
+
+  getExecution: function (transformationId) {
+    return getFetcher(`${URLs.DEBUG_EXECUTIONS}/${transformationId}`);
+  },
+
+  getExecutionName: function (id) {
+    return normaliseData(getFetcher(`${URLs.EXECUTION}/${id}`)).then((response) => {
+      return normaliseData(response);
+    });
+  },
+
+  getExecutionModules: function (transformationId) {
+    return getFetcher(`${URLs.DEBUG_EXECUTIONS}/${transformationId}/modules`);
+  },
+
+  compareExecutions: function (id1, id2) {
+    return getFetcher(`${URLs.DEBUG_EXECUTIONS}/${id1}/compare/${id2}`);
+  },
+
+  findVariableOrigin: function (id, pattern) {
+    return getFetcher(`${URLs.DEBUG_VARIABLE_ORIGIN}/${id}?variable=${pattern}`);
+  },
+
+  findTripleOrigin: function (id, pattern) {
```

Comment:
Please revise and refactor variables within this file so it is more understandable ... e.g. what is `id` and `pattern` ...